airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/partitions/types.py

```diff
@@ -25,4 +26,26 @@ def __init__(self, partition: Partition):
 """
 Typedef representing the items that can be added to the ThreadBasedConcurrentStream
 """
-QueueItem = Union[Record, Partition, PartitionCompleteSentinel, PartitionGenerationCompletedSentinel, Exception]
+QueueItem = Union[Record, Partition, PartitionCompleteSentinel, PartitionGenerationCompletedSentinel]
+
+
+class QueueItemObject:
+
+    def __init__(self, item: QueueItem):
+        self.value = item
+        self._priority = self._get_priority(self.value)
+
+    @staticmethod
+    def _get_priority(value) -> int:
+        # The order of the `isinstance` is a bit funky but it is order in terms of which QueueItem we expect to see the most
```

Comment:
should this raise an exception on unexpected type to ensure we don't forget to update the method if we ever add a new one?

Alternative would be to set a default priority.

edit: I like [this test ](https://github.com/airbytehq/airbyte/pull/34454/files#diff-112346e6a265b2833e32b22737c645d637935a4bab900d8dd581f158cd0e9d29R18) with the caveat that the current implement actually returns an `Optional[int]`

Reply:
I added the test as you were commenting. 

For the link, it seems like it is pointing to a blank like. Can you be more explicit about which method return `Optional[int]`?

Reply:
`_get_priority` does not return anything if the value is not of the expected type

Reply:
It's raising an exception though right? So the return type is indeed just `int` and not `Optional[int]`, right?

Reply:
I don't see where the method is raising an exception. As far as I can tell, the check relies on the unit test to verify all types of `QueueItem`s are handled. is that what you mean?

Reply:
What about [this line](https://github.com/airbytehq/airbyte/pull/34454/files#diff-47d1a1209ee8ebf3c79fab4d943369073681a3ed31421089cb33690baeeefa87R49)?